sabnzbd/interface.py

```diff
@@ -1078,6 +1104,37 @@ def saveGeneral(self, **kwargs):
         else:
             raise Raiser(self.__root)
 
+    @secured_expose(check_api_key=True, check_configlock=True)
+    def uploadConfig(self, **kwargs):
+        admin_backup = kwargs.get("admin_backup")
+
+        # Only accept the backup file if it can be opened as a zip archive and contains sabnzbd.ini
+        valid_backup = False
+        try:
+            # Read the zip file into RAM so that it can be closed and deleted before restart
+            backup_ref = io.BytesIO(admin_backup.file.read())
+            admin_backup.file.close()
+            admin_zip_ref = zipfile.ZipFile(backup_ref, "r")
+            for f in admin_zip_ref.infolist():
```

Comment:
See other comment, if we use a constant with all the expected filenames, we can just loop over that.